pilot/cmd/pilot-agent/main.go

```diff
@@ -212,12 +246,39 @@ var (
 				log.Info("Using existing certs")
 			}
 
-			sa := istio_agent.NewAgent(&proxyConfig, &istio_agent.AgentConfig{
+			secOpts := &security.Options{
 				PilotCertProvider:  pilotCertProvider,
-				JWTPath:            jwtPath,
 				OutputKeyCertToDir: outputKeyCertToDir,
+				JWTPath:            jwtPath,
 				ClusterID:          clusterIDVar.Get(),
-			})
+				FileMountedCerts:   fileMountedCertsEnv,
+				CAEndpoint:         caEndpointEnv,
+			}
+			secOpts.PluginNames = strings.Split(pluginNamesEnv, ",")
+
+			secOpts.EnableWorkloadSDS = true
+
+			secOpts.EnableGatewaySDS = enableGatewaySDSEnv
+			secOpts.CAProviderName = caProviderEnv
+
+			// TODO: extract from ProxyConfig
+			secOpts.TrustDomain = trustDomainEnv
+			secOpts.Pkcs8Keys = pkcs8KeysEnv
+			secOpts.ECCSigAlg = eccSigAlgEnv
+			secOpts.RecycleInterval = staledConnectionRecycleIntervalEnv
+			secOpts.ECCSigAlg = eccSigAlgEnv
+			secOpts.SecretTTL = secretTTLEnv
+			secOpts.SecretRotationGracePeriodRatio = secretRotationGracePeriodRatioEnv
+			secOpts.RotationInterval = secretRotationIntervalEnv
+			secOpts.InitialBackoffInMilliSec = int64(initialBackoffInMilliSecEnv)
+			// Disable the secret eviction for istio agent.
+			secOpts.EvictionDuration = 0
```

Comment:
why not assign these values in the above struct liternal?

Reply:
I just moved code around - the more you change the larger the PR and people start complaining :-)
Yes, we should, but in a separate PR. Probably combined with moving whatever is relevant to ProxyConfig.